### PR TITLE
Additional Integration tests

### PR DESCRIPTION
**Pull request by jan888adams**
This PR adds a new integration test for the newly introduced functionality.

## Summary by CodeRabbit

- **New Features**
  - Added comprehensive tests to verify cache tagging and invalidation behavior for assets, documents, and objects, including support for custom cache types and configuration-based enablement or disablement.
  - Introduced tests for additional cache tags in HTTP responses when various Pimcore elements are loaded.
  - Expanded test coverage for conditional cache invalidation and tagging based on element type, subtype, and configuration flags.

- **Bug Fixes**
  - Ensured that cache tags are not added or invalidated for disabled element types, subtypes (e.g., folders, images), or when global settings are off.

- **Tests**
  - Refactored and extended test factories for clearer and more specific test data creation.
  - Added new test scenarios for edge cases and configuration-driven behaviors in cache tagging and invalidation.

### CODE REVIEW

File: tests/Integration/Tagging/TagDocumentTest.php [hunk 1]
```diff
@@ -200,4 +200,30 @@ public function request_is_tagged_with_root_document_tag_when_loaded(): void
...
+     * @test
+     */
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'assets' => [
+                'types' => [
+                    'image' => false,
+                ],
+            ],
+        ],
```
Comment (jdreesen) [inline tests/Integration/Tagging/TagDocumentTest.php:214]: Shouldn't this be about documents?

File: tests/Integration/Invalidation/InvalidateObjectTest.php [hunk 2]
```diff
@@ -64,4 +69,144 @@ public function response_is_invalidated_when_object_is_deleted(): void
...
+
+    /**
+     * @test
+     */
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'objects' => true,
+        ],
+    ])]
+    public function response_is_not_invalidated_object_is_of_type_folder_on_update(): void
```
Comment (jdreesen) [inline tests/Integration/Invalidation/InvalidateObjectTest.php:81]:
```suggestion
    public function response_is_not_invalidated_when_object_is_of_type_folder_on_update(): void
```

File: tests/Integration/Invalidation/InvalidateObjectTest.php [hunk 3]
```diff
@@ -64,4 +69,144 @@ public function response_is_invalidated_when_object_is_deleted(): void
...
+
+    /**
+     * @test
+     */
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'objects' => true,
+        ],
+    ])]
+    public function response_is_not_invalidated_object_is_of_type_folder_on_delete(): void
```
Comment (jdreesen) [inline tests/Integration/Invalidation/InvalidateObjectTest.php:96]:
```suggestion
    public function response_is_not_invalidated_when_object_is_of_type_folder_on_delete(): void
```

File: tests/Integration/Invalidation/InvalidateDocumentTest.php [hunk 4]
```diff
@@ -64,4 +71,152 @@ public function response_is_invalidated_when_document_is_deleted(): void
...
+
+    /**
+     * @test
+     */
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'documents' => true,
+        ],
+    ])]
+    public function response_is_not_invalidated_when_document_is_from_type_hardlink_on_update(): void
```
Comment (jdreesen) [inline tests/Integration/Invalidation/InvalidateDocumentTest.php:83]:
```suggestion
    public function response_is_not_invalidated_when_document_is_of_type_hardlink_on_update(): void
```

File: tests/Integration/Invalidation/InvalidateDocumentTest.php [hunk 5]
```diff
@@ -64,4 +71,152 @@ public function response_is_invalidated_when_document_is_deleted(): void
...
+
+    /**
+     * @test
+     */
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'documents' => true,
+        ],
+    ])]
+    public function response_is_not_invalidated_when_document_is_from_type_hardlink_on_delete(): void
```
Comment (jdreesen) [inline tests/Integration/Invalidation/InvalidateDocumentTest.php:98]:
```suggestion
    public function response_is_not_invalidated_when_document_is_of_type_hardlink_on_delete(): void
```

File: tests/Integration/Invalidation/InvalidateDocumentTest.php [hunk 6]
```diff
@@ -64,4 +71,152 @@ public function response_is_invalidated_when_document_is_deleted(): void
...
+
+    /**
+     * @test
+     */
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'documents' => true,
+        ],
+    ])]
+    public function response_is_not_invalidated_when_document_is_from_type_email(): void
```
Comment (jdreesen) [inline tests/Integration/Invalidation/InvalidateDocumentTest.php:113]:
```suggestion
    public function response_is_not_invalidated_when_document_is_of_type_email_on_update(): void
```

File: tests/Integration/Invalidation/InvalidateDocumentTest.php [hunk 7]
```diff
@@ -64,4 +71,152 @@ public function response_is_invalidated_when_document_is_deleted(): void
...
+
+    /**
+     * @test
+     */
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'documents' => true,
+        ],
+    ])]
+    public function response_is_not_invalidated_when_document_is_from_type_email_on_delete(): void
```
Comment (jdreesen) [inline tests/Integration/Invalidation/InvalidateDocumentTest.php:128]:
```suggestion
    public function response_is_not_invalidated_when_document_is_of_type_email_on_delete(): void
```

File: tests/Integration/Invalidation/InvalidateDocumentTest.php [hunk 8]
```diff
@@ -64,4 +71,152 @@ public function response_is_invalidated_when_document_is_deleted(): void
...
+
+    /**
+     * @test
+     */
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'documents' => true,
+        ],
+    ])]
+    public function response_is_not_invalidated_when_document_is_from_type_folder_on_update(): void
```
Comment (jdreesen) [inline tests/Integration/Invalidation/InvalidateDocumentTest.php:143]:
```suggestion
    public function response_is_not_invalidated_when_document_is_of_type_folder_on_update(): void
```

File: tests/Integration/Invalidation/InvalidateDocumentTest.php [hunk 9]
```diff
@@ -64,4 +71,152 @@ public function response_is_invalidated_when_document_is_deleted(): void
...
+
+    /**
+     * @test
+     */
+    #[ConfigureExtension('neusta_pimcore_http_cache', [
+        'elements' => [
+            'documents' => true,
+        ],
+    ])]
+    public function response_is_not_invalidated_when_document_is_from_type_folder_on_delete(): void
```
Comment (jdreesen) [inline tests/Integration/Invalidation/InvalidateDocumentTest.php:158]:
```suggestion
    public function response_is_not_invalidated_when_document_is_of_type_folder_on_delete(): void
```

**Review by jdreesen — APPROVED** [review]

File: tests/Integration/Helpers/TestObjectFactory.php [hunk 10]
```diff
@@ -17,4 +19,27 @@ public static function simple(): TestDataObject
...
+
+    public static function simpleFolder(): DataObject\Folder
+    {
+        $folder = new DataObject\Folder();
+        $folder->setId(23);
+        $folder->setKey('test_folder');
+        $folder->setParentId(1);
+
+        return $folder;
+    }
```
Comment (jan888adams) [inline tests/Integration/Helpers/TestObjectFactory.php:44]: @jdreesen I believe we should consider passing the id as a parameter, along with content or key. This approach could enhance the readability of our tests. What are your thoughts?

Comment (jdreesen) [reply tests/Integration/Helpers/TestObjectFactory.php:44]: Sounds good. But maybe in a separate PR.

File: tests/Integration/Invalidation/InvalidateDocumentTest.php [hunk 11]
```diff
@@ -64,4 +71,152 @@ public function response_is_invalidated_when_document_is_deleted(): void
...
+
+    /**
+     * @test
+     */
+    public function response_is_not_invalidated_when_documents_are_disabled_on_delete(): void
+    {
+        $this->document->setKey('updated_test_document_page')->save();
+
+        $this->cacheManager->invalidateTags(Argument::any())->shouldNotHaveBeenCalled();
+    }
```
[the remark this answers is not in the document]
Comment (jdreesen) [reply tests/Integration/Invalidation/InvalidateDocumentTest.php:221]: @jan888adams isn't the bot right, and it doesn't test what the name says?

[the remark this answers is not in the document]
Comment (jdreesen) [reply tests/Integration/Invalidation/InvalidateDocumentTest.php:221]: Fixed in #23

File: tests/Integration/Invalidation/InvalidateDocumentTest.php [hunk 12]
```diff
@@ -64,4 +71,152 @@ public function response_is_invalidated_when_document_is_deleted(): void
...
+                ],
+            ],
+        ],
+    ])]
+    public function response_is_not_invalidated_when_document_type_is_disabled_on_delete(): void
+    {
+        $this->document->setKey('updated_test_document_page')->save();
+
+        $this->cacheManager->invalidateTags(Argument::any())->shouldNotHaveBeenCalled();
+    }
```
[the remark this answers is not in the document]
Comment (jdreesen) [reply tests/Integration/Invalidation/InvalidateDocumentTest.php:201]: Fixed in #23
